src/views/Submit.vue

```diff
@@ -145,6 +145,21 @@ export default {
 
 	mixins: [ViewsMixin],
 
+	/*
+	 * This is used to confirm that the user wants to leave the page
+	 * if the form is unsubmitted.
+	 */
+	beforeRouteUpdate(to, from, next) {
+		// This navigation guard is called when the route changes.
+		// continue with the navigation if there are no changes or the user confirms to leave the form
+		if (this.confirmLeaveForm()) {
+			next()
+		}
+
+		// Otherwise cancel the navigation
+		next(false)
+	},
```

Comment:
```suggestion
		// This navigation guard is called when the route parameters changed (e.g. form hash)
		// continue with the navigation if there are no changes or the user confirms to leave the form
		if (this.confirmLeaveForm()) {
			next()
		} else {
			// Otherwise cancel the navigation
			next(false)
		}
	},

	beforeRouteLeave(to, from, next) {
		// This navigation guard is called when the route changed and a new view should be shown
		// continue with the navigation if there are no changes or the user confirms to leave the form
		if (this.confirmLeaveForm()) {
			next()
		} else {
			// Otherwise cancel the navigation
			next(false)
		}
	},
```

After testing this PR: We need both route update and route leave. Otherwise not all cases are handled.